### ISSUE TITLE
rvm issues with fish rvm plugin

### ISSUE THREAD

**Issue by lythesia** [bpinto/oh-my-fish#404]
env

os: osx 10.10.2
fish: `fish, version 2.1.1-1206-gb4aa2b7`
rvm: `rvm 1.26.10 (master)`

Basically I can use rvm without obvious errors, but I find that newly created gemset does not inhreit from global, here comes related info:

newly-created gemset

```
$ rvm gemset use try
Using ruby-2.2.1 with gemset try

$ gem list

*** LOCAL GEMS ***

bigdecimal (1.2.6)
io-console (0.4.3)
json (1.8.1)
psych (2.0.8)
rake (10.4.2)
rdoc (4.2.0)

$ rvm info | grep GEM
    GEM_HOME:     "/Users/lythesia/.rvm/gems/ruby-2.2.1@try"
    GEM_PATH:     "/Users/lythesia/.rvm/gems/ruby-2.2.1@try/Users/lythesia/.rvm/gems/ruby-2.2.1@global" # <= this is strange
```

global gemset

```
$ rvm gemset use global
..

$ gem list

*** LOCAL GEMS ***

actionmailer (4.2.0)
actionpack (4.2.0)
actionview (4.2.0)
activejob (4.2.0)
activemodel (4.2.0)
activerecord (4.2.0)
activesupport (4.2.0)
(bunches of gems)
..

$ rvm info | grep GEM
    GEM_HOME:     "/Users/lythesia/.rvm/gems/ruby-2.2.1@global"
    GEM_PATH:     "/Users/lythesia/.rvm/gems/ruby-2.2.1@global"
```

And when I switch to bash, everything works well. Any advises? Thx!

**Comment by bucaran** [comment]
I use rbenv so I am not sure how to help you. Could it be a problem with `rvm` itself?

I don't think this has to do with user permissions, but just in case, you are not sudoing everytime you install a `gem` do you?

**Comment by lythesia** [comment]
@bucaran I think not. I tried bash and zsh, both works well with rvm. Surely I've never touched sudo during install rvm or gem.

**Comment by bucaran** [comment]
I didn't write the `rvm` plugin and I am not sure what the _intention_ of the plugin is.

If you can describe the specification for what the plugin should do I would gladly write a new one.

**Comment by lythesia** [comment]
[rvm](https://rvm.io) is a little "heavy" I think, and I've not read its source. Personally, I just use it to manager ruby/gem versions of apps(with gemsets).

I wonder  if any one encounter this would share his solution here. For now I use rbenv instead and it works well. I'll be very appreciate if this plugin can be fixed :)

**Comment by bucaran** [comment]
Like I said, if you could please write a simple spec of what you think a `rvm` plugin should do in addition to what the `rvm` standalone utility provides, please share it.

**Comment by lythesia** [comment]
Sorry, I mean the rvm itself not the rvm plugin for fish(it's short, right).

Personally I use these features frequently:

* install multiple ruby binaries and switch between them
* configure one default ruby for daily use
* create multiple gemsets handling different gems
* configure one default gemset also for daily use
* a global gemset can be inherited by others (as default), or more generally, specify some gemset inherit by others

Basically they all are functions of rvm, but original rvm is not compatible with fish, and this plugin for fish seems not work fine, so I report this issue.

**Comment by bpinto** [comment]
With #462 this issue should now be specific to the rbenv plugin.
